### PR TITLE
adds velocity drive mode & WPI competition changes

### PR DESCRIPTION
**Pull request by brhecker**
This branch contains the changes needed for the PID velocity drive mode. It replaces driving in brake mode for balancing.
The branch also has some tuning changes made for the WPI competition as we started working out of this branch in order to use velocity drive.

### CODE REVIEW

**Review by snowpuppy — CHANGES_REQUESTED** [review]
Overall, I love it. The one thing that I really want to see changed is the removal of the print statements. The print statements are going to get called a lot, which means the serial port will get spammed.

Question: Does the DriveFixedSpeed ever get used anywhere outside the dashboard? It doesn't look like it.

File: src/main/java/frc/robot/commands/Drive/DriveFixedSpeed.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.commands.Drive;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.Drive;
+import frc.robot.testingdashboard.TestingDashboard;
+
+public class DriveFixedSpeed extends CommandBase {
```
[the remark this answers is not in the document]
Comment (brhecker) [reply src/main/java/frc/robot/commands/Drive/DriveFixedSpeed.java:11]: The DriveFixedSpeed command can probably just be removed. It was just a quick thing intended for testing and tuning PID loops. Its not used anywhere

File: src/main/java/frc/robot/helpers/VelocityDriveSparkMax.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        m_driveMode = mode;
+    }
+
+    @Override
+    public void set(double speed)
+    {
+        if(m_driveMode == DriveMode.kPIDVelocity)
+        {
+            double rpm = speed * Constants.DRIVE_MAX_MOTOR_RPM;
+            System.out.println("Driving in Velocity PID mode at " + rpm + "rpm, from speed "+ speed + ", P:"+m_PidController.getP());
```
[the remark this answers is not in the document]
Comment (brhecker) [reply src/main/java/frc/robot/helpers/VelocityDriveSparkMax.java:65]: Ah, yes. This was just for debugging purposes, as we had an issue with PID values not getting set when we expected them to be. Fully agree it should be removed

File: src/main/java/frc/robot/subsystems/Drive.java [hunk 3]
```diff
@@ -245,6 +257,33 @@ public void toggleIdleMode() {
     }
   }
 
+  public void togglePIDDriveMode() {
+    if(m_driveMode == DriveMode.kPIDVelocity){
+      setDriveMode(DriveMode.kPower);
+    }
```
[the remark this answers is not in the document]
Comment (brhecker) [reply src/main/java/frc/robot/subsystems/Drive.java:263]: Okay, I will keep that in mind

**Pull request merged**